systems-management/nisysmgmt.yml

```diff
@@ -0,0 +1,2092 @@
+        description: Internal errors that occured in the service.
+        items:
+          $ref: '#/definitions/Error'
+
+  Operation:
+    description: An operation provided by the API
+    type: object
+    properties:
+      available:
+        type: boolean
+        description: Whether the operation is available to the caller.
+        example: true
+      version:
+        type: integer
+        format: int32
+        description: The version of the available operation.
+        example: 1
+    required: [available]
+
+  V1Operations:
+    description: V1 operations
+    type: object
+    properties:
+      operations:
+        description: >-
+          Available operations in the v1 version of the API:
+
+          - getJobs: The ability to retrieve the jobs.
+          
+          - getSystems: The ability to retrieve the systems information.
+
+          - getGroups: The ability to retrieve the groups.
+    
+          - manageSystems: The ability to register or unregister systems.
+          
+          - createOrCancelJobs: The ability to create or cancel jobs.
+          
+          - activateSystems: The ability to activate systems.
+          
+          - updateSystemMetadata: The ability to update systems metadata.
+          
+          - manageFeeds: The ability to configure feeds and retrieve the available packages.
+          
+          - createOrUpdateGroups: The ability to create, update, or delete groups.
+          
+        type: object
+        properties:
+          getJobs:
+            $ref: '#/definitions/Operation'
+          getSystems:
+            $ref: '#/definitions/Operation'
+          getGroups:
+            $ref: '#/definitions/Operation'
+          manageSystems:
+            $ref: '#/definitions/Operation'
+          createOrCancelJobs:
+            $ref: '#/definitions/Operation'
+          activateSystems:
+            $ref: '#/definitions/Operation'
+          updateSystemMetadata:
+            $ref: '#/definitions/Operation'
+          manageFeeds:
+            $ref: '#/definitions/Operation'
+          createOrUpdateGroups:
+            $ref: '#/definitions/Operation' 
+
+  QueryRequest:
+    description: Request for querying resources. More info here -> https://github.com/StefH/System.Linq.Dynamic.Core/wiki/Dynamic-Expressions and https://stefh.github.io/System.Linq.Dynamic.Core/html/144078bd-d1ad-bece-4daa-7fdd3a7ed7dc.htm
+    title: Query Request
+    properties:
+      skip:
+        description: The number of resources to skip.
+        type: integer
+        format: int64
+        default: 0
+        minimum: 0
+        example: 100
+      take:
+        description: The number of resources to return. The maximum value is 1000.
+        type: integer
+        format: int64
+        minimum: 0
+        maximum: 1000
+        example: 100
+      filter:
+        type: string
+        example: connected.data.state==CONNECTED
+        description: The criteria used to filter resources.
```

Comment:
For the filter, I recommend that our public APIs are explicit and describe both the syntax of the language and explicitly state the properties which can be used in the filter.

Take a look at the [Query-Assets](http://demo.systemlink.io/niapis/?urls.primaryName=Asset%20Performance%20Management#/assets/query-assets ) request and you can use the same text here too.